File: dmn-test-cases/standard/tck/1.3/cl3/0085-decision-services/translator/expected/java/dmn/DecisionService_014.java

import java.util.*;
import java.util.stream.Collectors;

@javax.annotation.Generated(value = {"ds.ftl", "decisionService_014"})
@com.gs.dmn.runtime.annotation.DRGElement(
    namespace = "",
    name = "decisionService_014",
    label = "",
    elementKind = com.gs.dmn.runtime.annotation.DRGElementKind.DECISION_SERVICE,
    expressionKind = com.gs.dmn.runtime.annotation.ExpressionKind.OTHER,
    hitPolicy = com.gs.dmn.runtime.annotation.HitPolicy.UNKNOWN,
    rulesCount = -1
)
public class DecisionService_014 extends com.gs.dmn.runtime.JavaTimeDMNBaseDecision<String> {
    public static final com.gs.dmn.runtime.listener.DRGElement DRG_ELEMENT_METADATA = new com.gs.dmn.runtime.listener.DRGElement(
        "",
        "decisionService_014",
        "",
        com.gs.dmn.runtime.annotation.DRGElementKind.DECISION_SERVICE,
        com.gs.dmn.runtime.annotation.ExpressionKind.OTHER,
        com.gs.dmn.runtime.annotation.HitPolicy.UNKNOWN,
        -1
    );

    private static class DecisionService_014LazyHolder {
        static final DecisionService_014 INSTANCE = new DecisionService_014();
    }
    public static DecisionService_014 instance() {
        return DecisionService_014LazyHolder.INSTANCE;
    }

    private final Decision_014_2 decision_014_2;

    private DecisionService_014() {
        this(new Decision_014_2());
    }

    private DecisionService_014(Decision_014_2 decision_014_2) {
        this.decision_014_2 = decision_014_2;
    }

    @java.lang.Override()
    public String applyMap(java.util.Map<String, String> input_, com.gs.dmn.runtime.ExecutionContext context_) {
        try {
            return apply(input_.get("inputData_014_1"), input_.get("decision_014_3"), context_);
        } catch (Exception e) {
            logError("Cannot apply decision 'DecisionService_014'", e);
            return null;
        }
    }

    @java.lang.Override()
    public String applyPojo(com.gs.dmn.runtime.ExecutableDRGElementInput input_, com.gs.dmn.runtime.ExecutionContext context_) {
        try {
            return apply(((DecisionService_014Input_)input_).getInputData_014_1(), ((DecisionService_014Input_)input_).getDecision_014_3(), context_);
        } catch (Exception e) {
            logError("Cannot apply element 'DecisionService_014'", e);
            return null;
        }
    }

    public String apply(String inputData_014_1, String decision_014_3, com.gs.dmn.runtime.ExecutionContext context_) {
        try {
            // Start DS 'decisionService_014'
            com.gs.dmn.runtime.annotation.AnnotationSet annotationSet_ = context_ != null ? context_.getAnnotations() : null;
            com.gs.dmn.runtime.listener.EventListener eventListener_ = context_ != null ? context_.getEventListener() : null;
            com.gs.dmn.runtime.external.ExternalFunctionExecutor externalExecutor_ = context_ != null ? context_.getExternalFunctionExecutor() : null;
            com.gs.dmn.runtime.cache.Cache cache_ = context_ != null ? context_.getCache() : null;
            long decisionService_014StartTime_ = System.currentTimeMillis();
            com.gs.dmn.runtime.listener.Arguments decisionService_014Arguments_ = new com.gs.dmn.runtime.listener.Arguments();
            decisionService_014Arguments_.put("inputData_014_1", inputData_014_1);
            decisionService_014Arguments_.put("decision_014_3", decision_014_3);
            eventListener_.startDRGElement(DRG_ELEMENT_METADATA, decisionService_014Arguments_);

            // Bind input decisions
            cache_.bind("decision_014_3", decision_014_3);

            // Evaluate DS 'decisionService_014'
            String output_ = lambda.apply(inputData_014_1, decision_014_3, context_);

            // End DS 'decisionService_014'
            eventListener_.endDRGElement(DRG_ELEMENT_METADATA, decisionService_014Arguments_, output_, (System.currentTimeMillis() - decisionService_014StartTime_));

            return output_;
        } catch (Exception e) {
            logError("Exception caught in 'decisionService_014' evaluation", e);
            return null;
        }
    }

    public com.gs.dmn.runtime.LambdaExpression<String> lambda =
        new com.gs.dmn.runtime.LambdaExpression<String>() {
            public String apply(Object... args_) {
                String inputData_014_1 = 0 < args_.length ? (String) args_[0] : null;
                String decision_014_3 = 1 < args_.length ? (String) args_[1] : null;
                com.gs.dmn.runtime.ExecutionContext context_ = 2 < args_.length ? (com.gs.dmn.runtime.ExecutionContext) args_[2] : null;
                com.gs.dmn.runtime.annotation.AnnotationSet annotationSet_ = context_ != null ? context_.getAnnotations() : null;
                com.gs.dmn.runtime.listener.EventListener eventListener_ = context_ != null ? context_.getEventListener() : null;
                com.gs.dmn.runtime.external.ExternalFunctionExecutor externalExecutor_ = context_ != null ? context_.getExternalFunctionExecutor() : null;
                com.gs.dmn.runtime.cache.Cache cache_ = context_ != null ? context_.getCache() : null;

                // Apply child decisions
                String decision_014_2 = DecisionService_014.this.decision_014_2.apply(inputData_014_1, context_);

                return decision_014_2;
            }
    };
}

File: dmn-test-cases/standard/tck/1.3/cl3/0085-decision-services/translator/expected/python/dmn/DecisionService_014.py
import typing
import decimal
import datetime
import isodate
import time

import jdmn.runtime.DefaultDMNBaseDecision
import jdmn.runtime.RuleOutput
import jdmn.runtime.RuleOutputList

import jdmn.runtime.annotation.Annotation
import jdmn.runtime.annotation.AnnotationSet
import jdmn.runtime.annotation.DRGElementKind
import jdmn.runtime.annotation.ExpressionKind
import jdmn.runtime.annotation.HitPolicy

import jdmn.runtime.cache.Cache

import jdmn.runtime.external.ExternalFunctionExecutor

import jdmn.runtime.listener.Arguments
import jdmn.runtime.listener.DRGElement
import jdmn.runtime.listener.EventListener
import jdmn.runtime.listener.Rule

import Decision_014_2


# Generated(value = {"ds.ftl", "decisionService_014"})
class DecisionService_014(jdmn.runtime.DefaultDMNBaseDecision.DefaultDMNBaseDecision):
    DRG_ELEMENT_METADATA: jdmn.runtime.listener.DRGElement.DRGElement = jdmn.runtime.listener.DRGElement.DRGElement(
        "",
        "decisionService_014",
        "",
        jdmn.runtime.annotation.DRGElementKind.DRGElementKind.DECISION_SERVICE,
        jdmn.runtime.annotation.ExpressionKind.ExpressionKind.OTHER,
        jdmn.runtime.annotation.HitPolicy.HitPolicy.UNKNOWN,
        -1
    )
    _instance = None

    def __init__(self):
        raise RuntimeError("Call instance() instead")

    @classmethod
    def instance(cls):
        if cls._instance is None:
            cls._instance = cls.__new__(cls)
            jdmn.runtime.DefaultDMNBaseDecision.DefaultDMNBaseDecision.__init__(cls._instance)
            cls._instance.initSubDecisions()
        return cls._instance

    def initSubDecisions(self, decision_014_2: Decision_014_2.Decision_014_2 = None):
        self.decision_014_2 = Decision_014_2.Decision_014_2() if decision_014_2 is None else decision_014_2

    def apply(self, inputData_014_1: typing.Optional[str], decision_014_3: typing.Optional[str], annotationSet_: jdmn.runtime.annotation.AnnotationSet.AnnotationSet, eventListener_: jdmn.runtime.listener.EventListener.EventListener, externalExecutor_: jdmn.runtime.external.ExternalFunctionExecutor.ExternalFunctionExecutor, cache_: jdmn.runtime.cache.Cache.Cache) -> typing.Optional[str]:
        try:
            # Start DS 'decisionService_014'
            decisionService_014StartTime_ = int(time.time_ns()/1000)
            decisionService_014Arguments_ = jdmn.runtime.listener.Arguments.Arguments()
            decisionService_014Arguments_.put("inputData_014_1", inputData_014_1)
            decisionService_014Arguments_.put("decision_014_3", decision_014_3)
            eventListener_.startDRGElement(self.DRG_ELEMENT_METADATA, decisionService_014Arguments_)

            # Bind input decisions
            cache_.bind("decision_014_3", decision_014_3)

            # Evaluate DS 'decisionService_014'
            output_: typing.Optional[str] = self.evaluate(inputData_014_1, decision_014_3, annotationSet_, eventListener_, externalExecutor_, cache_)

            # End DS 'decisionService_014'
            eventListener_.endDRGElement(self.DRG_ELEMENT_METADATA, decisionService_014Arguments_, output_, (int(time.time_ns()/1000) - decisionService_014StartTime_))

            return output_
        except Exception as e:
            self.logError("Exception caught in 'decisionService_014' evaluation", e)
            return None
    def evaluate(self, inputData_014_1: typing.Optional[str], decision_014_3: typing.Optional[str], annotationSet_: jdmn.runtime.annotation.AnnotationSet.AnnotationSet, eventListener_: jdmn.runtime.listener.EventListener.EventListener, externalExecutor_: jdmn.runtime.external.ExternalFunctionExecutor.ExternalFunctionExecutor, cache_: jdmn.runtime.cache.Cache.Cache) -> typing.Optional[str]:
        # Apply child decisions
        decision_014_2: typing.Optional[str] = self.decision_014_2.apply(inputData_014_1, annotationSet_, eventListener_, externalExecutor_, cache_)

        return decision_014_2
